### ISSUE TITLE
FXIOS-1880 ⁃ Firefox crashes when scaning a specified QR Code

### ISSUE THREAD

**Issue by KLszTsu** [mozilla-mobile/firefox-ios#8567]
## Steps to reproduce

Scan this QR Code:
<details>
<summary><b>Image:</b></summary>

![IMG_9144](https://user-images.githubusercontent.com/12217600/119445386-40dea880-bd1c-11eb-8aad-325e209bb54f.jpg)

</details>

~~1. Generate a QR Code that contains information other than URL, for example a string of plain text.~~
~~2. Scan the qrcode using Firefox for iOS.~~

### Expected behavior

Firefox would tell the user that firefox can't handle the QR Code and then show the user the content of the qrcode.

### Actual behavior

Firefox crashed.

---

Also as a side note, I met the bug because I scanned a QR Code with `SMSTO:` link. As iOS builtin Camera app and Android builtin Google Lens as well as a number of social media apps can recognize the link, I think Firefox should recognize more types of links as well.

The other crashes that happened when Firefox scaned QR Codes containing plain text are not reproducible, thus removed from the issue.

### Device & build information
* Device: iOS 14.4.2 (18D70)
* Build version: Firefox Daylight 33.1 (4629)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-2045)

**Comment by gutley** [comment]
@KuangLei - Do you have an example QR Code that crashes. I tried a variety here without issue in the same version of Firefox. 

E.g This one just contains the text "Some text":
<img width="178" alt="image" src="https://user-images.githubusercontent.com/2040690/119320563-a32f9e80-bc73-11eb-9116-9ce321857c80.png">

**Comment by KuangLei** [comment]
The QR Code that I initially met still crashes Firefox. But I have no ideas what other QR Codes I had tried that day that also led to a crash was.

<details>
<summary><b>Image:</b></summary>

![IMG_9144](https://user-images.githubusercontent.com/12217600/119445386-40dea880-bd1c-11eb-8aad-325e209bb54f.jpg)
</details>

<details>
<summary><b>Recording:</b></summary>

[IMG_9144](https://user-images.githubusercontent.com/12217600/119444673-25bf6900-bd1b-11eb-9aa6-e22c5d14b7ab.MP4)
</detail>

**Comment by SimonBasca** [comment]
This issue can be reproduce by scanning this screenshot of the video.

<img width="291" alt="image" src="https://user-images.githubusercontent.com/32130829/119464598-8ff00200-bd4b-11eb-98ea-b51f3c82e363.png">

**Comment by gutley** [comment]
Thanks - I can confirm that that crashes.

Developers - This happens because:

That QR code equates to the text:
```
SMSTO:1922:場所代碼：2421 4972 3343 106\n本簡訊是簡訊實聯制發送，限防疫目的使用。
```

When it gets to here:
```swift
    func BrowserViewController.didScanQRCodeWithText(_ text: String) {
        TelemetryWrapper.recordEvent(category: .action, method: .scan, object: .qrCodeText)
        let content = TextContentDetector.detectTextContent(text)
        switch content {
        case .some(.link(let url)):
            UIApplication.shared.open(url, options: [:], completionHandler: nil)
        case .some(.phoneNumber(let phoneNumber)):
            let url = URL(string: "tel:\(phoneNumber)")!
            UIApplication.shared.open(url, options: [:], completionHandler: nil)
        default:
            guard let tab = tabManager.selectedTab else { return }
            submitSearchText(text, forTab: tab)
        }
    }
}

````

The TextContentDetector resolves it to being a phone number with text `2421 4972 3343 106`. (Note the spaces)

It then incorrectly assumes it can create a URL from this using ` let url = URL(string: "tel:\(phoneNumber)")!` and then force unwrapping it. But you can't create such a URL with spaces in it so the force unwrap is what crashes.

There are several things wondering with this code in general:

1. For any code that goes through the .phoneNumber case here, we need to remove all whitespace from the text before trying to create the URL.
2. We need to account for the URL not being able to be created instead of force unwrapping it
3. SMSTO: URLs are being wrongly handled completely.

Looking at what the standard iOS QR code reader does in this situation:
 It offers to create a message to 1922
![IMG_3651](https://user-images.githubusercontent.com/2040690/119464841-0f290a00-bd3b-11eb-8c60-a0869cf032b3.jpg)

Accepting that you end up in Messages 
![IMG_3649](https://user-images.githubusercontent.com/2040690/119465030-35e74080-bd3b-11eb-9b6e-d4a7aa92d6d9.PNG)

At the least we should stop the crash and maybe support SMSTO: URLS properly

**Comment by gutley** [comment]
Also, translating that text gives: "
Venue code: 2421 4972 3343 106\nThis newsletter is sent by the newsletter and is limited to epidemic prevention."

So it was never a phone number in the first place (not that text detection would know that).

**Comment by KuangLei** [comment]
`1922` is Taiwan's "Communicable Disease Reporting and Consultation Hotline". The Centers for Disease Control requires citizens to send sms like this to the hotline to record everyone who enters a store or shopping mall. The code `2421 4972 3343 106` is the code for the mall I'm visiting.

**Comment by SimonBasca** [comment]
Verifying as fix on main 60640b8ab7182502cdcd1d94749edb01db807a05.
